lib/Models/data_controller.dart

```diff
@@ -0,0 +1,44 @@
+// ignore_for_file: avoid_print
+
+import 'package:covert_html_to_pdf/dio_helper.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'ProductModel.dart';
+
+final getDataFuture =
+    ChangeNotifierProvider<GetDataFromApi>((ref) => GetDataFromApi());
+
+class GetDataFromApi extends ChangeNotifier {
+  ProductModel listDataModel = ProductModel();
+
+  GetDataFromApi() {
+    getData();
+  }
+
+  Future getData() async {
+    DioHelper.getData(url: "products").then((value) {
```

Comment:
for best practice of dart team prefer to use 'await' with store return value in variable  keyword instead of using 'then'